Project.ipynb

```diff
@@ -149,10 +153,64 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "def add_temporal_dim(A,n_updates):\n",
+    "## disable some links for a certain amount of time\n",
+    "##\n",
+    "## modes:\n",
+    "## light mode: few links, short time\n",
+    "## heavy mode: few links, long time\n",
+    "## unstable mode: many links, short time\n",
+    "## extreme mode: many links, long time\n",
+    "##\n",
+    "## priority: random, near, far\n",
+    "## prioritize disabling of links involving nodes near or far from Earth\n",
+    "## \n",
+    "## custom_n_targets = number of targets\n",
+    "## custom_offtimes: array containing possible values for off times expressed in number of updates of A\n",
+    "\n",
+    "def disable_links(A,n_updates, mode=\"light\",priority=\"random\", custom_n_targets = None, custom_offtimes = None):\n",
+    "\n",
+    "    arrays = [A for i in range(n_updates)]\n",
+    "    At = np.stack(arrays,axis=2)      #adds temporal dimension\n",
+    "    \n",
+    "    t_nodes = np.arange(n)\n",
+    "    \n",
+    "    if priority == \"random\": np.random.shuffle(t_nodes)   #nodes targeted for link removal\n",
+    "    elif priority == \"near\": t_nodes = sorted_nodes[:int(np.ceil(n/3))]\n",
+    "    elif priority == \"far\": t_nodes = np.flip(sorted_nodes)[:int(np.ceil(n/3))]\n",
     "    \n",
-    "    At = np.broadcast_to(A.reshape((n,n,1)),(n,n,n_updates))\n",
-    "    ### At = ???(At) ##rimuovere link disattivati\n",
+    "    possible_links = []\n",
+    "    possible_links = np.array([[x,y] for x in t_nodes for y in range(n) if x!=y if A[x,y]<tau_max if [y,x] not in possible_links])\n",
+    "\n",
+    "    if mode == \"light\":  \n",
+    "        n_targets = possible_links.shape[0]//3\n",
+    "        offtimes = [n_updates//10]   # selected links stay off for around 1/10 of the total time\n",
+    "            \n",
+    "    if mode == \"heavy\":\n",
+    "        n_targets = possible_links.shape[0]//3\n",
+    "        offtimes = [n_updates//3]\n",
+    "        \n",
+    "    if mode == \"unstable\":\n",
+    "        n_targets = possible_links.shape[0]*3//4\n",
+    "        offtimes = [n_updates//10]\n",
+    "        \n",
+    "    if mode == \"extreme\":\n",
+    "        n_targets = possible_links.shape[0]*3//4\n",
+    "        offtimes = [n_updates//3]\n",
+    "        \n",
```

Comment:
What if mode is None of these?
i suggest using light mode as default, so if the mode name is uncorrect you can throw an exception or silently run light mode